kgx/source/rdf_source.py

```diff
@@ -273,7 +273,7 @@ def triple(self, s: URIRef, p: URIRef, o: URIRef) -> None:
             # treating predicate as an edge
             self.add_edge(s, o, p)
 
-        if len(self.edge_cache) >= self.CACHE_SIZE:
+        if self.edge_cache and len(self.edge_cache) <= self.CACHE_SIZE:
```

Comment:
The idea behind this if clause is to:
- test if edge_cache is greater than CACHE_SIZE. If it is greater than CACHE_SIZE then flush the cache by processing records in edge_cache.

Reply:
@deepakunni3, I understand the intent of the first bullet point. That said, given how the `edge_cache `is populated, and the size of `CACHE_SIZE`, I wonder if this block of code will ever be called?  I'm also now sure if the second bullet point is always properly done. Is there also a DRY issue here in both places. Sorry... been a few days since I looked at all of this, but when I did, the test behaviour seemed unusual.

Maybe after we diagnose the real sources of the unit test failures: are they reproducible in your hands? If not, then I should perhaps reinstall all my dependencies from scratch in case there is something stale in my testing environment. If so, then what has changed in the code base (or the critical semantic inputs, e.g. LinkML, BMT, Biolink Model?) to break the tests?

Reply:
The 2nd bullet point is done in the `parse` method of RdfSource where after the entire file is read, any remaining records in node and edge cache are processed. It would be a major bug if the parse method is not getting to the records left behind in the cache.

Regarding the tests - All unit tests seem to work fine. Perhaps a fresh install (and virtual env) will help?

Reply:
> Regarding the tests - All unit tests seem to work fine. Perhaps a fresh install (and virtual env) will help?

@deepakunni3  (cc: @sierra-moxon) I cloned the **master** branch and built a fresh `venv `and `pip install .`, on my Mac OSX laptop, but I am still getting a small handful of RDF / OWL unit test errors. Please kindly advise.